cardano-cli/src/Cardano/CLI/Shelley/Parsers.hs

```diff
@@ -433,86 +433,92 @@ pKeyCmd =
+    , subParser "sign"
+        (Opt.info pTransactionSign $ Opt.progDesc "Sign a transaction")
+    , subParser "witness"
+        (Opt.info pTransactionCreateWitness $ Opt.progDesc "Create a transaction witness")
+    , subParser "assemble"
+        (Opt.info pTransactionAssembleTxBodyWit
+          $ Opt.progDesc "Assemble a tx body and witness(es) to form a transaction")
+    , pSignWitnessBackwardCompatible
+    , subParser "submit"
+        (Opt.info pTransactionSubmit . Opt.progDesc $
+           mconcat
+             [ "Submit a transaction to the local node whose Unix domain socket "
+             , "is obtained from the CARDANO_NODE_SOCKET_PATH enviromnent variable."
+             ]
+          )
+    , subParser "calculate-min-fee"
+        (Opt.info pTransactionCalculateMinFee $ Opt.progDesc "Calulate the min fee for a transaction")
+    , subParser "txid"
+        (Opt.info pTransactionId $ Opt.progDesc "Print a transaction identifier")
+    ]
+ where
+  subParser :: String -> ParserInfo TransactionCmd -> Parser TransactionCmd
+  subParser name pInfo = Opt.subparser $ Opt.command name pInfo
+
+  assembleInfo :: ParserInfo TransactionCmd
+  assembleInfo =
+    Opt.info pTransactionAssembleTxBodyWit
+      $ Opt.progDesc "Assemble a tx body and witness(es) to form a transaction"
+
+  pSignWitnessBackwardCompatible :: Parser TransactionCmd
+  pSignWitnessBackwardCompatible =
+    Opt.subparser
+      $ Opt.command "sign-witness" assembleInfo <> Opt.internal
+
+  pTransactionBuild :: Parser TransactionCmd
+  pTransactionBuild = TxBuildRaw <$> some pTxIn
+                                 <*> some pTxOut
+                                 <*> pTxTTL
+                                 <*> pTxFee
+                                 <*> many pCertificateFile
+                                 <*> many pWithdrawal
+                                 <*> pTxMetadataJsonSchema
+                                 <*> many pMetaDataFile
+                                 <*> optional pUpdateProposalFile
+                                 <*> pTxBodyFile Output
+
+  pTransactionSign  :: Parser TransactionCmd
+  pTransactionSign = TxSign <$> pTxBodyFile Input
+                            <*> pSomeWitnessSigningData
+                            <*> optional pNetworkId
+                            <*> pTxFile Output
+
+  pTransactionCreateWitness :: Parser TransactionCmd
+  pTransactionCreateWitness = TxCreateWitness
+                                <$> pTxBodyFile Input
+                                <*> pWitnessSigningData
+                                <*> optional pNetworkId
+                                <*> pOutputFile
+
+  pTransactionAssembleTxBodyWit :: Parser TransactionCmd
+  pTransactionAssembleTxBodyWit = TxAssembleTxBodyWitness
+                                    <$> pTxBodyFile Input
+                                    <*> some pWitnessFile
+                                    <*> pOutputFile
```

Comment:
Not a fan of this style of alignment, but this is how most of them were before so :man_shrugging: